data-serving/data-service/src/controllers/case.ts

```diff
@@ -301,41 +298,35 @@ export class CasesController {
                 });
                 return;
             }
-            const bulkWriteResult = await Case.bulkWrite(
-                // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                req.body.cases.map((c: any) => {
-                    if (
-                        c.caseReference?.sourceId &&
-                        c.caseReference?.sourceEntryId
-                    ) {
-                        return {
-                            updateOne: {
-                                filter: {
-                                    'caseReference.sourceId':
-                                        c.caseReference.sourceId,
-                                    'caseReference.sourceEntryId':
-                                        c.caseReference.sourceEntryId,
-                                },
-                                update: c,
-                                upsert: true,
-                            },
-                        };
-                    } else {
-                        return {
-                            insertOne: {
-                                document: c,
-                            },
-                        };
-                    }
-                }),
-                { ordered: false },
-            );
+            // Use this method rather than Case.bulkWrite() as that method
```

Comment:
Any thoughts as to why this method seems to be easier on memory? `bulkWrite` seems to be the recommended API for this (re: [tip](https://docs.mongodb.com/manual/reference/method/db.collection.initializeUnorderedBulkOp/)), but I assume you held everything else constant and witnessed OOMs with bulkWrite vs. this method.

Reply:
Yes everything else held constant it was OOM. The initializeUnorderedBulkOp method breaks it into much smaller groups. It also does schema validation which bulkWrite does not do (which caused the need for all the peripheral changes in this PR) 

From the docs of bulkWrite: 
The number of operations in each group cannot exceed the value of the maxWriteBatchSize of the database. As of MongoDB 3.6, this value is 100,000. This value is shown in the isMaster.maxWriteBatchSize field.

This limit prevents issues with oversized error messages. If a group exceeds this limit, the client driver divides the group into smaller groups with counts less than or equal to the value of the limit. For example, with the maxWriteBatchSize value of 100,000, if the queue consists of 200,000 operations, the driver creates 2 groups, each with 100,000 operations.
https://docs.mongodb.com/manual/reference/method/db.collection.bulkWrite/#execution-of-operations

From the docs of db.collection.initializeUnorderedBulkOp: 
Each group of operations can have at most 1000 operations. If a group exceeds this limit, MongoDB will divide the group into smaller groups of 1000 or less. For example, if the bulk operations list consists of 2000 insert operations, MongoDB creates 2 groups, each with 1000 operations.
https://docs.mongodb.com/manual/reference/method/db.collection.initializeUnorderedBulkOp/#execution-of-operations

Reply:
Got it -- so we're avoiding the second most likely place to OOM, which is the Mongo heap.